## Первая метрика 


In [2]:
import pandas as pd
import numpy as np

In [4]:
df = pd.read_csv('tips.csv')

In [5]:
df.head(3)

total_bill   tip     sex smoker  day    time  size  price_per_person  \
0       16.99  1.01  Female     No  Sun  Dinner     2              8.49   
1       10.34  1.66    Male     No  Sun  Dinner     3              3.45   
2       21.01  3.50    Male     No  Sun  Dinner     3              7.00   

           Payer Name         CC Number Payment ID  
0  Christy Cunningham  3560325168603410    Sun2959  
1      Douglas Tucker  4478071379779230    Sun4608  
2      Travis Walters  6011812112971322    Sun4458

In [6]:
df['day'].unique()

array(['Sun', 'Sat', 'Thur', 'Fri'], dtype=object)

In [7]:
days_of_week = ["Thur", "Fri", "Sat", "Sun"]
daily_tips_sum = {}

for day in days_of_week:
    # Фильтрация строк DataFrame для текущего дня недели.
    # df['day'] == day создает булеву маску.
    df_day_filtered = df[df['day'] == day]
    
    # Подсчет общей суммы чаевых для отфильтрованных строк (т.е. для текущего дня).
    # Используем метод .sum() на столбце 'tip' отфильтрованного DataFrame.
    total_tips_for_day = df_day_filtered['tip'].sum()
    
    daily_tips_sum[day] = total_tips_for_day

In [12]:
for day, total_tips in daily_tips_sum.items():
    print(f"{day} - {total_tips:.2f}")

Thur - 171.83
Fri - 51.96
Sat - 260.40
Sun - 247.39


- Thur - 171.83
- Fri - 51.96
- Sat - 260.40
- Sun - 247.39

## Вторая метрика 

In [14]:
days_of_week = ["Thur", "Fri", "Sat", "Sun"]
daily_tips_sum = {}
for day in days_of_week:
    df_day_filtered = df[df['day'] == day]
    total_tips_for_day = df_day_filtered['tip'].sum()
    daily_tips_sum[day] = total_tips_for_day

print("\n--- Метрика 2: Процентное соотношение чаевых по дням недели ---")

total_all_tips = df['tip'].sum()


percentage_daily_tips = {}

if total_all_tips > 0: 
    for day in days_of_week: 
        sum_tips_for_day = daily_tips_sum.get(day, 0) 
        percentage = (sum_tips_for_day / total_all_tips) * 100
        percentage_daily_tips[day] = percentage
else:
    
    for day in days_of_week:
        percentage_daily_tips[day] = 0.0

print("\nПроцентное соотношение чаевых по дням недели:")
for day, perc in percentage_daily_tips.items():
    print(f"{day} - {perc:.2f}%")




--- Метрика 2: Процентное соотношение чаевых по дням недели ---

Процентное соотношение чаевых по дням недели:
Thur - 23.49%
Fri - 7.10%
Sat - 35.59%
Sun - 33.82%




Процентное соотношение чаевых по дням недели:
- Thur - 23.49%
- Fri - 7.10%
- Sat - 35.59%
- Sun - 33.82%


## Третья метрика 

In [15]:
print("\n--- Метрика 3: Три компании посетителей с наибольшей суммой чаевых ---")

unique_sizes = df['size'].value_counts().index.tolist()
size_tips_sum = {}

for size_val in unique_sizes: # Renamed 'size' to 'size_val' to avoid conflict with df.size
    df_size_filtered = df[df['size'] == size_val]
    total_tips_for_size = df_size_filtered['tip'].sum()
    size_tips_sum[size_val] = total_tips_for_size

sorted_size_tips = sorted(size_tips_sum.items(), key=lambda item: item[1], reverse=True)
top_3_companies = sorted_size_tips[:3]

print("\nТри компании посетителей с наибольшей суммой чаевых:")
if len(top_3_companies) >= 1:
    print(f"Группа из {top_3_companies[0][0]} человек - {top_3_companies[0][1]:.2f}")
if len(top_3_companies) >= 2:
    print(f"Группа из {top_3_companies[1][0]} человек - {top_3_companies[1][1]:.2f}")
if len(top_3_companies) >= 3:
    print(f"Группа из {top_3_companies[2][0]} человек - {top_3_companies[2][1]:.2f}")
elif len(top_3_companies) < 3 and len(top_3_companies) > 0:
     print("В данных меньше трех уникальных размеров групп.")
elif len(top_3_companies) == 0:
     print("Нет данных о чаевых для групп.")



--- Метрика 3: Три компании посетителей с наибольшей суммой чаевых ---

Три компании посетителей с наибольшей суммой чаевых:
Группа из 2 человек - 402.84
Группа из 4 человек - 153.01
Группа из 3 человек - 128.94


Три компании посетителей с наибольшей суммой чаевых:
- Группа из 2 человек - 402.84
- Группа из 4 человек - 153.01
- Группа из 3 человек - 128.94

## Четвертая метрика 